src/gui/shell.cpp

```diff
@@ -1217,29 +1218,45 @@ void Shell::wheelEvent(QWheelEvent *ev)
 	if (!m_attached || !m_mouseEnabled) {
 		return;
 	}
-#ifdef Q_OS_MAC
-	// For some reason <ScrollWheel*> scrolls multiple lines at once
-	// we have to account for it, to make sure that pixelDelta() is used correctly.
-	const int scroll_columns = 6;
-	const int scroll_rows = 3;
-	// Minimal scroll step
-	int scroll_step_x = cellSize().width() * scroll_columns;
-	int scroll_step_y = cellSize().height() * scroll_rows;
-	// Total scroll delta considering previous events
-	QPoint total_delta = m_mouse_wheel_delta_fraction + ev->pixelDelta();
-	// Delta rounded to a minimal scroll step
-	QPoint cell_delta(total_delta.x() / scroll_step_x, total_delta.y() / scroll_step_y);
-	// Save remainder for future events
-	m_mouse_wheel_delta_fraction = total_delta - QPoint(cell_delta.x() * scroll_step_x, cell_delta.y() * scroll_step_y);
-
-	int horiz = cell_delta.x();
-	int vert = cell_delta.y();
+#if (QT_VERSION < QT_VERSION_CHECK(5, 5, 0))
+	int defaultDeltasPerStep = 120;
 #else
-	int horiz, vert;
-	horiz = ev->angleDelta().x();
-	vert = ev->angleDelta().y();
+	int defaultDeltasPerStep = QWheelEvent::DefaultDeltasPerStep;
 #endif
 
+	QPointF scroll_delta_f(float(ev->angleDelta().x()) / defaultDeltasPerStep,
+						   float(ev->angleDelta().y()) / defaultDeltasPerStep);
+
+#if (QT_VERSION >= QT_VERSION_CHECK(5, 7, 0))
+	if (ev->inverted()) {
+		scroll_delta_f = -scroll_delta_f;
+	}
+#endif
+
+	// Reset scroll remainder if we change scroll direction;
+	int direction_x = int(scroll_delta_f.x() < 0.0f ? -1 :
+						  scroll_delta_f.x() > 0.0f ? 1 : m_scroll_last_direction.x());
```

Comment:
You should remove the C-Style cast here, it will only suppress valid compile-time issues.

Also, this line is hard to follow.  A small `static` helper function might make this easier to read given the line below is almost identical.

Reply:
I've moved it into a `normalize` function